wordpress-mobile/WordPress-iOS
# Post Sign-up: Fix incorrect overlay shown in post sign-up flow on WordPress

Fixes #21749
Depends on #21751
## Description
This PR fixes an issue where users attempting to create a .com site in the post-sign-up flow are presented with two consecutive overlays. 

## Testing Instructions

1. Delete both WordPress and Jetpack to prevent any shared or cached data
2. Install WordPress
3. Create a new account
4. Background the app, do not close it
5. Confirm your new account's email on the same device
6. You should be navigated back to the WordPress app automatically
7. Tap "Done"
8. Tap "Create a new site"
9. ✅ Make sure that a site creation-specific modal is presented
10. Dismiss the modal
11. ✅ Make sure that you are still on the post-sign-up screen
12. Tap "Not right now"
13. ✅ Make sure that a fullscreen modal is presented
14. Dismiss the modal
15. ✅  Make sure that you land on My Site with the tab bar & Jetpack features hidden

## Regression Notes
1. Potential unintended areas of impact
Site creation after sign up flow in Jetpack

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
